packages/components/src/components/hds/popover-primitive/index.js

```diff
@@ -0,0 +1,334 @@
+   * @method classNames
+   * @return {string} The "class" attribute to apply to the root element
+   */
+  get classNames() {
+    let classes = ['hds-popover-primitive'];
+
+    // add a class based on the @isInline argument
+    if (this.isInline) {
+      classes.push('hds-popover-primitive--is-inline');
+    } else {
+      classes.push('hds-popover-primitive--is-block');
+    }
+
+    return classes.join(' ');
+  }
+
+  @action
+  showPopover() {
+    this.popoverElement.showPopover();
+  }
+
+  @action
+  hidePopover() {
+    this.popoverElement.hidePopover();
+  }
+
+  @action
+  togglePopover() {
+    this.popoverElement.togglePopover();
+  }
+
+  // fired just _before_ the "popover" is shown or hidden
+  @action
+  onBeforeTogglePopover(event) {
+    if (event.newState === 'closed') {
+      // we need this flag to check if it's in the "closing" process,
+      // because the browser automatically returns the focus to the "trigger" button
+      // and this would re-open immediately the popover because of the `focusin` event
+      this.isClosing = true;
+    }
+  }
+
+  // fired just _after_ the "popover" is shown or hidden
+  @action
+  onTogglePopover(event) {
+    if (event.newState === 'open') {
+      this.isOpen = true;
+
+      // we call the "onOpen" callback if it exists (and is a function)
+      let { onOpen } = this.args;
+      if (typeof onOpen === 'function') {
+        onOpen();
+      }
+    } else {
+      this.isOpen = false;
+      // reset the "isClosing" flag (the `toggle` event is fired _after_ the popover is closed)
+      this.isClosing = false;
+
+      // if the popover was initially forced to be open (using the "manual" state) then revert its status to `auto` once the user interacts with it
+      if (this.args.isOpen) {
+        this.popoverElement.popover = 'auto';
+      }
+
+      // we call the "onClose" callback if it exists (and is a function)
+      let { onClose } = this.args;
+      if (typeof onClose === 'function') {
+        onClose();
+      }
+    }
+  }
+
+  @action
+  onMouseEnter() {
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+    this.showPopover();
+  }
+
+  @action
+  onFocusIn() {
+    // don't re-open the popover if the focus is returned because the closing
+    if (!this.isClosing) {
+      if (this.timer) {
+        clearTimeout(this.timer);
+      }
+      this.showPopover();
+    }
+  }
+
+  @action
+  onMouseLeave() {
+    this.timer = setTimeout(this.hidePopover.bind(this), 500);
+  }
+
+  @action
+  onFocusOut(event) {
+    // TODO! discuss with Alex if/why we need this check here
+    // due to inconsistent implementation of relatedTarget across browsers we use the activeElement as a fallback
+    // if the related target is not part of the disclosed content we close the disclosed container
+    if (
+      !this.containerElement.contains(
+        event.relatedTarget || document.activeElement
+      )
+    ) {
+      this.hidePopover();
+    }
```

Comment:
This logic was set in place in the `MenuPrimitive` to ensure the revealed content disappears only when the focus is moved away from the component (trigger or content). If you remove this check you will not be able to tab into the popover content; the focus will move to the next element on the page.

### With this check
![PopoverPrimitive-focusout-logic](https://github.com/hashicorp/design-system/assets/788096/4b570699-23dd-4e00-955f-35c52b89a147)


### Without this check
![PopoverPrimitive-focusout--no-logic](https://github.com/hashicorp/design-system/assets/788096/c8bcb73c-08ab-412b-b6d7-3a9db4244c3f)